Tests/Parse/invalid-data-decl.silt

```diff
@@ -1,12 +1,13 @@
 -- RUN: %silt --verify parse %s
 module DataDecl where
 
-data Foo : Set where
+data Foo : Type where
   | Foo : nat -> nat -> Foo
   | Bar : nat -> Foo
 
-data Bar : Set where
--- FIXME: This error is awful.
--- expected-error @1 {{expected declaration}}
-| Bar : nat -> nat -> Bar
-| Baz : nat -> Bar
+data Bar : Type where
+| Bar : nat -> nat -> Bar -- expected-error {{data constructors may only appear within the scope of a data declaration}}
```

Comment:
Maybe a note that looks at indentation level and suggests indenting? 🤔

We could keep track of the current decl kind being parsed